Source/DafnyCore/AST/Expressions/Expressions.cs

```diff
@@ -986,6 +986,10 @@ public DatatypeValue(IToken tok, string datatypeName, string memberName, List<Ex
   }
 
   public IToken NameToken => tok;
+  public bool SetIndent(int indentBefore, TokenNewIndentCollector formatter) {
+    formatter.SetMethodLikeIndent(StartToken, OwnedTokens, indentBefore);
```

Comment:
This method is the "default" formatting heuristic, which can take into account type parameters, named arguments, etc.